### DISCUSSION TITLE
Making a GIF from a spritesheet

### DISCUSSION THREAD

**Discussion by MeCryptoCookie** [ImageMagick/ImageMagick#6081] [Help] [answered]
### ImageMagick version

7.1.0-62

### Operating system, version and so on

macOS Ventura 13.2

### Description

I'm trying to turn a 512x512 PNG file into a 64x64 GIF file that contains all the 64x64 frames in the original image in sequence from top left to bottom right. Every file has 8 rows of 8 frames. I've tried various versions of convert, but most of them end up in an error `geometry does not contain image` and an output image that contains at most an 8-frame GIF of the top row of the original image. What am I missing here?

E.g. for what I tried and figured should work:
```
magick convert -dispose 3 -delay 10 -loop 0 sheet.png -crop 64x64 +repage anim.gif
```

**Comment by snibgo** [comment, marked as answer]
In future please provide all the items so we can reproduce the problem, such as the input file.

Your command works fine for me with a sample PNG.

If your input PNG has a canvas offset, that could cause the problem. Insert "`+repage`" after reading the input. Does that fix it?

You have "`magick convert`" which gives you the old v6 syntax instead of the new v7 syntax. Why do you do this? I suggest using just "`magick`" to use the new v7 syntax.

**Reply by MeCryptoCookie** [reply]
Apologies, I'm new to all of this Magick, so going off of commands I found on the internet which are mostly quite old. `+repage` after the input file name does seem to do the trick! Don't ask me how or what, because I still don't quite understand, but thanks regardless of that :D
